modules/apps/object/object-rest-impl/src/main/java/com/liferay/object/rest/internal/dto/v1_0/converter/ObjectEntryDTOConverter.java

```diff
@@ -121,6 +121,29 @@ public ObjectEntry toDTO(
 			objectEntry);
 	}
 
+	private void _addNestedFields(
+		String nestedFields, Map<String, Object> map, Object object,
+		String objectFieldName, ObjectRelationship objectRelationship) {
+
+		String[] nestedFieldsList = nestedFields.split(",");
+
+		String objectFieldNameNestedField = StringUtil.replaceLast(
+			objectFieldName.substring(
+				objectFieldName.lastIndexOf(StringPool.UNDERLINE) + 1),
+			"Id", "");
+
+		for (String nestedField : nestedFieldsList) {
+			if (nestedField.contains(objectFieldNameNestedField)) {
```

Comment:
Are we keeping the old "magic" functionality then?

Reply:
Well, as you know, we decide to keep the old "magical" functionality

Reply:
:face_in_clouds: